src/main/java/org/spongepowered/api/item/inventory/ItemStack.java

```diff
@@ -139,6 +141,20 @@ static ItemStack of(ItemType itemType, int quantity) {
         Builder quantity(int quantity) throws IllegalArgumentException;
 
         /**
+         * Adds a {@link Key} and related {@link Object} value to apply to the
+         * resulting {@link ItemStack}. Note that the resulting
+         * {@link ItemStack} may not actually accept the provided {@code Key}
+         * for various reasons due to support or simply that the value itself
+         * is not supported.
```

Comment:
What happens if the keyValue isnt supported?
Will they be discarded or will they cause an Exception during build()?